From 1858f4d6adb14623e77061d09da3541686eb02f2 Mon Sep 17 00:00:00 2001
From: Wang Quanyang <quanyang.wang@windriver.com>
Date: Fri, 12 Jul 2019 05:44:55 -0400
Subject: [PATCH] gpu: mail-400: change return value of mali_mem_vma_fault

driver/src/devicedrv/mali/linux/mali_memory.c:186:11: error: initialization of 'vm_fault_t (*)(struct vm_fault *)' {aka 'unsigned int (*)(struct vm_fault *)'} from incompatible pointer type 'int (*)(struct vm_fault *)' [-Werror=incompatible-pointer-types]
|   186 |  .fault = mali_mem_vma_fault,
|       |           ^~~~~~~~~~~~~~~~~~

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 linux/mali_memory.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/linux/mali_memory.c b/linux/mali_memory.c
index b0eb3c4..b422422 100644
--- a/linux/mali_memory.c
+++ b/linux/mali_memory.c
@@ -71,7 +71,7 @@ static void mali_mem_vma_close(struct vm_area_struct *vma)
 }
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
-static int mali_mem_vma_fault(struct vm_fault *vmf)
+static vm_fault_t mali_mem_vma_fault(struct vm_fault *vmf)
 #else
 static int mali_mem_vma_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
 #endif
-- 
2.22.0

